Build failed; the context from the [latest run](https://ci.jenkins.io/job/Infra/job/evergreen/job/PR-30/1/display/redirect) is:
<details><summary>Expand to view</summary>

```
make: *** [lint] Error 1
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Verifications)
Stage 'Verifications' skipped due to earlier failure(s)
[Pipeline] parallel
[Pipeline] [Client] { (Branch: Client)
[Pipeline] [Services] { (Branch: Services)
[Pipeline] [Client] stage
[Pipeline] [Client] { (Client)
[Pipeline] [Services] stage
[Pipeline] [Services] { (Services)
Stage 'Client' skipped due to earlier failure(s)
Stage 'Services' skipped due to earlier failure(s)
[Pipeline] [Client] }
[Pipeline] [Services] }
[Pipeline] [Client] // stage
[Pipeline] [Services] // stage
[Pipeline] [Client] }
[Pipeline] [Services] }
[Pipeline] // parallel
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build jenkins/evergreen)
Stage 'Build jenkins/evergreen' skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Test jenkins/evergreen)
Stage 'Test jenkins/evergreen' skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Publish jenkins/evergreen)
Stage 'Publish jenkins/evergreen' skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline

GitHub has been notified of this commit’s build result

ERROR: script returned exit code 2
Finished: FAILURE

```
</details>

[Powered by the Comment Logger](https://github.com/jenkins-infra/community-functions/tree/master/comment-logger)

--- a/Dockerfile
+++ b/Dockerfile
@@ -93,4 +93,4 @@ COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf
 CMD /usr/bin/supervisord -c /etc/supervisor/conf.d/supervisord.conf
 
 WORKDIR $EVERGREEN_HOME
-USER jenkins
+USER $user

--- a/Dockerfile
+++ b/Dockerfile
@@ -24,10 +24,6 @@ RUN mkdir -p /usr/share/jenkins/ref/ && \
     mkdir ${EVERGREEN_HOME}/jenkins/ && \
     mkdir ${JENKINS_HOME}
 
-# Jenkins directory is a volume, so configuration and build history
-# can be persisted and survive image upgrades
-VOLUME ${EVERGREEN_HOME}
-
 # for main web interface:
 EXPOSE ${http_port}
 # will be used by attached agents:
@@ -36,7 +32,8 @@ EXPOSE ${agent_port}
 
 # FIXME REMOVE (to ease iteration/speed just for now), see also shim-startup-wrapper.sh
 RUN wget --quiet https://updates.jenkins.io/download/war/2.107.1/jenkins.war -O /usr/share/jenkins/jenkins.war
-RUN apk add --no-cache curl # used by shim-startup-wrapper.sh
+RUN apk add --no-cache curl aria2 # used by shim-startup-wrapper.sh
+COPY scripts/casc-dependencies.aria /casc-dependencies.aria
 
 # Add the system dependencies for running Jenkins effectively
 #
@@ -72,7 +69,6 @@ RUN addgroup -g ${gid} ${group} \
 RUN mkdir -p /usr/local/bin
 COPY build/jenkins.sh /usr/local/bin/
 COPY build/jenkins-support /usr/local/bin/
-COPY build/install-plugins.sh /usr/local/bin/
 COPY scripts/shim-startup-wrapper.sh /usr/local/bin
 
 # Prepare the evergreen-client configuration
@@ -85,7 +81,16 @@ COPY build/configuration-as-code/target/configuration-as-code.hpi /usr/share/jen
 COPY jenkins-configuration.yaml /usr/share/jenkins/ref/jenkins.yaml
 ENV CASC_JENKINS_CONFIG=$JENKINS_HOME/jenkins.yaml
 
+RUN chown -R $user:$group $EVERGREEN_HOME
+
+# Jenkins directory is a volume, so configuration and build history
+# can be persisted and survive image upgrades
+VOLUME ${EVERGREEN_HOME}
+
 # Ensure the supervisord configuration is copied and executed by default such
 # that the Jenkins and evergreen-client processes both execute properly
 COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf
 CMD /usr/bin/supervisord -c /etc/supervisor/conf.d/supervisord.conf
+
+WORKDIR $EVERGREEN_HOME
+USER jenkins

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ check: lint
 	$(MAKE) -C services $@
 	$(MAKE) container-check
 
-container-prereqs: build/jenkins-support build/jenkins.sh build/install-plugins.sh scripts/shim-startup-wrapper.sh build/configuration-as-code/target/configuration-as-code.hpi
+container-prereqs: build/jenkins-support build/jenkins.sh scripts/shim-startup-wrapper.sh build/configuration-as-code/target/configuration-as-code.hpi
 
 container-check: shunit2 ./tests/tests.sh container
 	./tests/tests.sh
@@ -55,11 +55,6 @@ build/jenkins.sh:
 build/jenkins-support:
 	mkdir -p build
 	$(DOWNLOAD) $(SCRIPTS_URL)/jenkins-support > $@
-	chmod +x $@
-
-build/install-plugins.sh:
-	mkdir -p build
-	$(DOWNLOAD)  $(SCRIPTS_URL)/install-plugins.sh > $@
 	chmod +x $@
 
 build/configuration-as-code:

--- a/scripts/casc-dependencies.aria
+++ b/scripts/casc-dependencies.aria
@@ -1,0 +1,16 @@
+# https://updates.jenkins.io/experimental/latest/configuration-as-code.hpi # built by us
+https://updates.jenkins.io/2.107/latest/workflow-multibranch.hpi
+https://updates.jenkins.io/2.107/latest/job-dsl.hpi
+https://updates.jenkins.io/2.107/latest/workflow-api.hpi
+https://updates.jenkins.io/2.107/latest/workflow-cps.hpi
+https://updates.jenkins.io/2.107/latest/workflow-job.hpi
+https://updates.jenkins.io/2.107/latest/workflow-scm-step.hpi
+https://updates.jenkins.io/2.107/latest/workflow-step-api.hpi
+https://updates.jenkins.io/2.107/latest/workflow-support.hpi
+https://updates.jenkins.io/2.107/latest/branch-api.hpi
+https://updates.jenkins.io/2.107/latest/cloudbees-folder.hpi
+https://updates.jenkins.io/2.107/latest/scm-api.hpi
+https://updates.jenkins.io/2.107/latest/script-security.hpi
+https://updates.jenkins.io/2.107/latest/structs.hpi
+https://updates.jenkins.io/2.107/latest/ace-editor.hpi
+https://updates.jenkins.io/2.107/latest/jquery-detached.hpi

--- a/scripts/shim-startup-wrapper.sh
+++ b/scripts/shim-startup-wrapper.sh
@@ -31,6 +31,13 @@ fi
 # not following essentials.yaml declaration
 # On purpose to make the startup faster
 # anyway the whole file is a *shim* :P
-install-plugins.sh configuration-as-code:experimental
+download_tmp=/tmp/download-casc-plugins$RANDOM
+mkdir "$download_tmp"
+cd $download_tmp
+aria2c -x 4 -i /casc-dependencies.aria
+mkdir -p "$JENKINS_HOME/plugins/"
+mv ./*.hpi "$JENKINS_HOME/plugins/"
+rm -rf $download_tmp
+cd "$JENKINS_HOME"
 
 exec jenkins.sh "$@"

--- a/tests/tests.sh
+++ b/tests/tests.sh
@@ -39,10 +39,22 @@ test_docker_CLI_available() {
   assertEquals "expected message about daemon unavailable" 0 $?
 }
 
+# JENKINS-49861
 test_no_executor() {
   numExecutors=$( docker exec $container_under_test cat "$JENKINS_HOME/config.xml" | \
       grep '<numExecutors>0</numExecutors>' | tr -d ' ' )
   assertEquals "<numExecutors>0</numExecutors>" "$numExecutors"
+}
+
+# JENKINS-50195
+test_not_root() {
+  username=$( docker exec $container_under_test whoami )
+  assertEquals "jenkins" "$username"
+
+  for process_user in $( docker exec $container_under_test ps -o user | grep -v USER)
+  do
+    assertEquals "jenkins" "$process_user"
+  done
 }
 
 . ./shunit2/shunit2

